### ISSUE TITLE
Missing API: fs.rm, fsPromises.rm

### ISSUE THREAD

**Issue by BlackGlory** [streamich/memfs#677]
- [fs.rm](https://nodejs.org/api/fs.html#fs_fs_rm_path_options_callback)
- [fsPromsies.rm](https://nodejs.org/api/fs.html#fs_fspromises_rm_path_options)

**Comment by PatrickMennen** [comment]
This is not true however, it seems that it HAS been implemented correctly on fs.promises but the function is missing on fs

```ts
import { fs, vol } from 'memfs';

jest.mock('fs', () => fs);
jest.mock('fs/promises', () => fs.promises);
```

<img width="673" alt="image" src="https://user-images.githubusercontent.com/7114798/220088630-1dc076b9-f79b-4e3d-b7ce-336c69f2db78.png">

as you can see the function is correctly implemented on the promises library, but undefined in the non-promise version
